### PR TITLE
Also Look for the Amazon cred header to remove auth header

### PR DESCRIPTION
**Pull request by GaryCaldwell**
Looks like the amazon dual auth problem is now on other urls than amazonaws.com because github releases is using it now too

### CODE REVIEW

**Comment by GaryCaldwell** [comment]
The CI failures seem to be unrelated and failing on all PRs.

**Comment by 9oelM** [comment]
Some people are desperately waiting for this.. 😢

**Review by lsitest — CHANGES_REQUESTED** [review]
There is no need for two nested ifs for the delete condition.
The second one which has the condition : 
"(parsedNewUrl.hostname.endsWith(".amazonaws.com") || parsedNewUrl.searchParams.has("X-Amz-Credential"))", is enough to get things working again.

**Comment by hypermoose** [comment]
@lsitest Your likely right but I didnt want to regress anything so I was being conservative

**Comment by leonetosoft** [comment]
Complicated, updating everyone is immensely complicated! Many users who do not know how to update manual! what if the update is critical?

**Comment by 9oelM** [comment]
Please help thousands of poor and lost souls throughout the world by merging this PR. Who can approve this PR?

**Review by Chrisbright10 — APPROVED** [review]
looks great!

**Comment by Chrisbright10** [comment]
> There is no need for two nested ifs for the delete condition.
> The second one which has the condition :
> "(parsedNewUrl.hostname.endsWith(".amazonaws.com") || parsedNewUrl.searchParams.has("X-Amz-Credential"))", is enough to get things working again.

He didn't add any "if"s, he only modified the one that was already there. no reason to change it and potentially break functionality.

**Comment by Bernix01** [comment]
So is this happening?

**Review by mmaietta — APPROVED** [review]

**Comment by 9oelM** [comment]
The number of the poor souls is only increasing in number as the time goes. What it really takes is just to approve a commit of 7 lines of difference, one of which is a new line addition. 6 lines. I really hope that someone with the right to approve this PR exercises it at the very right time, and be the hero of the world just by pressing a few sequence of simple buttons.

**Comment by adelphes** [comment]
@Chrisbright10 @mmaietta - can you please merge this in so we can all move on with our lives.

It's been 18 days since updates were [reported](https://github.com/electron-userland/electron-builder/issues/5595) broken - this PR, which presumably will fix it, is 19 days old.
It's a one-line change with 2 approvals.

What are the electron-builder maintainers waiting for?

**Comment by mmaietta** [comment]
I'm really sorry @adelphes, I wish I could help more. I'm actually not a maintainer and don't have write access ☹️  I'm hoping to change that though.

I've opened https://github.com/electron-userland/electron-builder/pull/5630 to fix master branch so that (I'm presuming) it'd unblock a pre-release cut after this PR is merged. 

@develar can you please assist here?

**Comment by adelphes** [comment]
@mmaietta thanks very much and apologies - as a reviewer, I'd assumed you have write access.
Thanks for bumping this.

**Pull request merged**
